cmd/queue.go

```diff
@@ -0,0 +1,449 @@
+
+	// DELETE
+	deleteCommand()
+
+	// ENQUEUE
+	enqueueCommand()
+
+	// DEQUEUE
+	dequeueCommand()
+}
+
+var queueCmd = &cobra.Command{
+	Use:     "queue",
+	Aliases: []string{"q", "queues"},
+	Short:   "Create and manage queues",
+	Long:    `Queues are used to store messages in a sequential order. You can create, read, update, and delete queues using this command. You can also add and remove messages from queues.`,
+}
+
+func listCommand() {
+	listQueueCmd := &cobra.Command{
+		Use:   "list",
+		Short: "List existing queues",
+		RunE:  runListQueueCmd,
+	}
+
+	queueCmd.AddCommand(listQueueCmd)
+	addListFlags(listQueueCmd)
+}
+
+func readCommand() {
+	readQueueCmd := &cobra.Command{
+		Use:     "get [flags] queue_id",
+		Aliases: []string{"read"},
+		Short:   "Fetch details about a single queue",
+		Long: `Fetch details about a single queue. You must provide the ID of the queue you want to read.
+
+An example of reading a queue:
+
+  ivcap queue get urn:ivcap:queue:714e549b-ebab-5dd8-8ebd-2e4b0af76167`,
+		Args: validateReadCommandArgs,
+		RunE: runReadQueueCmd,
+	}
+
+	queueCmd.AddCommand(readQueueCmd)
+}
+
+func createCommand() {
+	createQueueCmd := &cobra.Command{
+		Use:   "create [flags] name",
+		Short: "Create a new queue",
+		Long: `Create a new queue with the specified name. Optionally, you can also provide a description and a policy for the queue.
+
+An example of creating a queue with a description:
+
+  ivcap queue create --description "This is a test queue" test_queue`,
+		Args: validateCreateQueueArgs,
+		RunE: runCreateQueueCmd,
+	}
+
+	queueCmd.AddCommand(createQueueCmd)
+	createQueueCmd.Flags().StringP("description", "d", "", "Description of the queue")
+	addPolicyFlag(createQueueCmd)
+}
+
+func deleteCommand() {
+	deleteQueueCmd := &cobra.Command{
+		Use:     "delete [flags] queue_id",
+		Aliases: []string{"remove"},
+		Short:   "Delete a queue",
+		Long: `Delete a queue with the specified ID. This action is irreversible. All messages in the queue will be lost.
+An example of deleting a queue:
+
+  ivcap queue delete test_queue`,
+		Args: validateDeleteCommandArgs,
+		RunE: runDeleteQueueCmd,
+	}
+	queueCmd.AddCommand(deleteQueueCmd)
+}
+
+func enqueueCommand() {
+	enqueueCmd := &cobra.Command{
+		Use:   "enqueue [flags] schema queue_id message",
+		Short: "Enqueue a message to a queue",
+		Long: `Enqueue a message to the specified queue. The message content must be in JSON format. You must also provide the schema of the message content.
+
+An example of enqueuing a message to a queue:
+
+  ivcap queue enqueue urn:ivcap:schema:queue:message.1 urn:ivcap:queue:714e549b-ebab-5dd8-8ebd-2e4b0af76167 "{\"temperature\": \"21\", \"location\": \"Buoy101\", \"timestamp\": \"2024-05-20T14:30:00Z\"}"`,
+		Args: validateEnqueueArgs,
+		RunE: runEnqueueCmd,
+	}
+
+	queueCmd.AddCommand(enqueueCmd)
+}
+
+func dequeueCommand() {
+	dequeueCmd := &cobra.Command{
+		Use:   "dequeue [flags] queue_id",
+		Short: "Dequeue messages from a queue",
```

Comment:
Where are the messages written to? Again, you may want to add a `-f` flag to set the destination. `-f -` would be stdout. In the case of multiple messages, the `-f` may need to point to a directory. Not sure if you would want then to write file names to stdout.

Thinking about it, maybe without a `-f` you use a table (which you may have implemented here) with the message content as a column ...

Reply:
> Where are the messages written to? Again, you may want to add a `-f` flag to set the destination. `-f -` would be stdout. In the case of multiple messages, the `-f` may need to point to a directory. Not sure if you would want then to write file names to stdout.
> 
> Thinking about it, maybe without a `-f` you use a table (which you may have implemented here) with the message content as a column ...

I have implemented the standard ivcap-cli console output options. The default option is to print to the console using a table. If json/yaml are chosen then the response is outputted to the console in the desired format.

Adding a `-f` option sounds good for long-running options. Maybe I could add an `-f` option for a single file and a `-d` option for a directory?